### PR TITLE
OHRM5X-1098: Attendance - Connect employee attendance screens with API

### PR DESCRIPTION
**Pull request by Super-Chama**

### CODE REVIEW

File: symfony/client/src/core/util/helper/datefns.ts [hunk 1]
```diff
@@ -182,6 +293,31 @@ const setClockInterval = (callback: (args: void) => void, interval = 1000) => {
...
 
+/**
+ * guessTimezone will first try to guess the current timezone name using
+ * ES6 Intl API. in the offchance it's not possible it will revert value using
+ * default timezone list.
+ * @typedef {Object} Timezone
+ * @property {string} label - timezone's english name
+ * @property {number} offset - timezone's offset in hours
+ */
+const guessTimezone = () => {
```
Comment (Super-Chama) [inline symfony/client/src/core/util/helper/datefns.ts:304]: @samanthajayasinghe @RajithaKumara this function will try to guess the timezone name. as a fallback i've added some hardcoded timezone names as in 4x.

File: symfony/plugins/orangehrmAttendancePlugin/Controller/AttendanceMockController.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Attendance\Controller;
+
+use OrangeHRM\Core\Controller\AbstractController;
+use OrangeHRM\Framework\Http\Request;
+use OrangeHRM\Framework\Http\Response;
+
+class AttendanceMockController extends AbstractController
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Controller/AttendanceMockController.php:27]: @SandeepaOHRM will delete this, once connect all FE features with BE APIs

File: symfony/plugins/orangehrmAttendancePlugin/Controller/EmployeePunchInOutController.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+class EmployeePunchInOutController extends AbstractVueController
+{
+    use AttendanceServiceTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function preRender(Request $request): void
+    {
+        if ($request->query->has('employeeId')) {
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Controller/EmployeePunchInOutController.php:39]:
```suggestion
        if ($request->query->has('empNumber')) {
```
better not to mess with `employeeId`

File: symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml [hunk 4]
```diff
@@ -51,12 +51,12 @@ apiv2_attendance_employee_attendance_record_summary:
   defaults:
     _api: OrangeHRM\Attendance\Api\EmployeeAttendanceSummaryAPI
 
-apiv2_attendance_timezones:
-  path: /api/v2/attendance/timezones
-  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
-  methods: [ GET ]
-  defaults:
-    _api: OrangeHRM\Attendance\Api\TimezonesAPI
+# apiv2_attendance_timezones:
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml:54]: @SandeepaOHRM please check this

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**